crates/sqlite_store/schema/schema_0.sql

```diff
@@ -0,0 +1,65 @@
+-- schema version control
+CREATE TABLE version (version INTEGER) STRICT;
+INSERT INTO version VALUES (1);
+
+-- network is the valid network for all other table data
+CREATE TABLE network
+(
+    name TEXT UNIQUE NOT NULL
+) STRICT;
+
+-- keychain is the json serialized keychain structure as JSONB,
+-- descriptor is the complete descriptor string,
+-- descriptor_id is a sha256::Hash id of the descriptor string w/o the checksum,
+-- last revealed index is a u32
+CREATE TABLE keychain
+(
+    keychain      BLOB PRIMARY KEY NOT NULL,
+    descriptor    TEXT UNIQUE      NOT NULL,
+    descriptor_id BLOB UNIQUE      NOT NULL,
+    last_revealed INTEGER
+) STRICT;
+
+-- hash is block hash hex string,
+-- block height is a u32,
+CREATE TABLE block
+(
+    hash   TEXT PRIMARY KEY NOT NULL,
+    height INTEGER          NOT NULL
+) STRICT;
+
+-- txid is transaction hash hex string (reversed)
+-- whole_tx is a consensus encoded transaction,
+-- last seen is a u64 unix epoch seconds
+CREATE TABLE tx
+(
+    txid      TEXT PRIMARY KEY NOT NULL,
+    whole_tx  BLOB             NOT NULL,
+    last_seen INTEGER
```

Comment:
Another good catch, I changed `tx.whole_tx` to be `NULL`-able and changed `insert_txs()` and `update_last_seen()` to be "UPSERT" meaning they `INSERT` a new `tx` record if the txid doesn't exist, otherwise they `UPDATE` either the `whole_tx` or `last_seen`.  I also updated `select_txs()` to only select tx where  `tx.whole_tx` is not `NULL`.